src/types/index.ts

```diff
@@ -71,3 +71,14 @@ export interface PaginationData {
   page: number;
   pageSize: number;
 }
+
+export interface PersonAttribute {
+  id: number;
+  label: string;
+  value: string;
+}
+
+export interface SearchByProps {
+  onSubmit: (searchParams: SearchParams, queryDescription: string) => void;
+  isLoading: boolean;
```

Comment:
Make on submit a promise. So that you don't need another prop to check the loading state.